pinot-core/src/main/java/org/apache/pinot/core/segment/processing/framework/SegmentProcessorFramework.java

```diff
@@ -72,11 +73,34 @@ public SegmentProcessorFramework(List<RecordReader> recordReaders, SegmentProces
       File workingDir)
```

Comment:
Let's add `@deprecated` annotation and add the comment that we prefer the other constructor over this.

Reply:
Added